cmd/root.go

```diff
@@ -44,22 +50,43 @@ func Execute() {
 
 func newOkta(flags *Flags) *okta.Okta {
 	o := okta.New(nil, flags.Debug)
-	o.BaseDomain = flags.BaseDomain
+	if flags.ClientID == "" {
+	    o.ClientID, o.ClientSecret, o.BaseDomain = ParseConfig(flags.InputConfig, flags.Username)
+	} else {
+	    o.ClientID = flags.ClientID
+        o.ClientSecret = flags.ClientSecret
+        o.BaseDomain = flags.BaseDomain
+	}
 	o.BindAddr = flags.BindAddr
-	o.ClientID = flags.ClientID
-	o.ClientSecret = flags.ClientSecret
 	o.Debug = flags.Debug
+	o.KubeConfig = flags.KubeConfig
+	o.Username = flags.Username
 	return o
 }
 
+func ParseConfig(path string, username string) (string, string, string) {
+
+	jsonConfig, err := ioutil.ReadFile(path)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	id, _, _, _ := jsonparser.Get(jsonConfig, username, "id")
+	secret, _, _, _ := jsonparser.Get(jsonConfig, username, "secret")
+	uri, _, _, _ := jsonparser.Get(jsonConfig, username, "uri")
+
+	return string(id), string(secret), string(uri)
+}
+
 func init() {
-	RootCmd.Flags().StringVar(&flags.ClientID, "client-id", "", "OAuth2 client ID of this application. (required)")
-	RootCmd.MarkFlagRequired("client-id")
-	RootCmd.Flags().StringVar(&flags.ClientSecret, "client-secret", "", "OAuth2 client secret of this application. (required)")
-	RootCmd.MarkFlagRequired("client-secret")
+	RootCmd.Flags().StringVar(&flags.ClientID, "client-id", "", "OAuth2 client ID of this application.")
+	RootCmd.Flags().StringVar(&flags.ClientSecret, "client-secret", "", "OAuth2 client secret of this application.")
 
-	RootCmd.PersistentFlags().StringVar(&flags.BaseDomain, "base-domain", "", "URL of the OpenID Connect issuer. (required)")
-	RootCmd.MarkPersistentFlagRequired("base-domain")
+	RootCmd.PersistentFlags().StringVar(&flags.BaseDomain, "base-domain", "", "URL of the OpenID Connect issuer.")
 	RootCmd.PersistentFlags().StringVar(&flags.BindAddr, "bind-addr", "127.0.0.1:8888", "HTTP address to listen at.")
 	RootCmd.PersistentFlags().BoolVar(&flags.Debug, "debug", false, "Raise log level to debug.")
+
+	RootCmd.PersistentFlags().StringVar(&flags.KubeConfig, "kubeconfig", "", "Path to the kubeconfig you want to update.")
+	RootCmd.PersistentFlags().StringVar(&flags.InputConfig, "config", "cluster_config.json", "Path to a json file containing the required keys/tokens. (see README)")
```

Comment:
You shouldn't have a default value set to cluster_config.json. I think this is breaking cli in terms of error handling. 
Also take into account that currently only required flag(s) are "base-domain", "client-id", "client-secret". 

I would expect that you also need to better handle parsing of the file in newOkta() function.